app/src/main/java/ch/epfl/polybazaar/SalesOverview.java

```diff
@@ -2,29 +2,186 @@
 
 import android.content.Intent;
 import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.Gravity;
 import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.ScrollView;
+import android.widget.TableRow;
 import android.widget.TextView;
 
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.constraintlayout.widget.ConstraintLayout;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+import ch.epfl.polybazaar.database.callback.LiteListingCallback;
+import ch.epfl.polybazaar.database.callback.LiteListingListCallback;
+import ch.epfl.polybazaar.litelisting.LiteListing;
+
+import static ch.epfl.polybazaar.litelisting.LiteListingDatabase.fetchLiteListing;
+import static ch.epfl.polybazaar.litelisting.LiteListingDatabase.fetchLiteListingList;
 
 public class SalesOverview extends AppCompatActivity {
 
+    // TODO: add tests for error cases (ex empty list on database)
+
+    private List<String> IDList;
+    private List<LiteListing> liteListingList;
```

Comment:
Might want to consider using a separate class for this in the future. Could be like a class that is filled as data is loaded and can be read once the data is ready. But that's just something to keep in mind for future expansion; right now, this works just fine.